src/LIBSVM.jl

```diff
@@ -143,6 +142,25 @@ function instances2nodes{U<:Real}(instances::AbstractMatrix{U})
     (nodes, nodeptrs)
 end
 
+function instances2nodes!{U<:Real}(nodes, nodeptrs, instances::AbstractMatrix{U})
+    nfeatures = size(instances, 1)
+    ninstances = size(instances, 2)
+    @assert size(nodes) == (nfeatures+1,ninstances)
```

Comment:
`@assert` is meant to check for the programmer's mistakes, not the user's. Use `size(nodes) == (nfeatures+1,ninstances) || throw(DimensionMismatch("size of nodes $(size(nodes) does not match expected size ($(nfeatures+1),$(ninstances))` or similar for these kinds of checks.